### DISCUSSION TITLE
How can I maintain text highlighting while scrolling?

### DISCUSSION THREAD

**Discussion by azzamsa** [wezterm/wezterm#4762] [Q&A] [answered]
### What Operating System(s) are you running on?

Linux Wayland

### Which Wayland compositor or X11 Window manager(s) are you using?

Gnome

### WezTerm version

20230712-072601-f4abf8fd

### Ask your question!

Hi, 👋🏼

I'm transitioning from Zellij, and I'm accustomed to holding the left click and scrolling the wheel simultaneously to select text without using the `next page / prev page` command.

In Wezterm, I'd like to replicate this behavior, but I haven't had any luck so far. This is particularly difficult when using [Bacon](https://dystroy.org/bacon/), as the error spans multiple pages, and I want to copy the entire content.

In Wezterm, the highlighting disappears right after I scroll the wheel.

**Comment by wez** [comment, marked as answer]
The selection highlight is preserved until the screen underneath it is repainted.

If you are using a full-screen tui like `tmux`, `zellij`, `vim` and so on, scrolling with the mouse wheel often is translated to the application repainting the screen for itself, which will clear the highlighted section on the screen.

Selecting while scrolling the wheel works fine for me here in raw wezterm.  I suspect that you are using a tui multiplexer inside wezterm, and that that is the heart of your issue.

**Reply by azzamsa** [reply]
It works fine in raw Wezterm.

https://github.com/wez/wezterm/assets/17734314/30d49239-534d-4043-ab13-5ff6863bbe2f

However, the same functionality is not present in [Bacon](https://dystroy.org/bacon/).

https://github.com/wez/wezterm/assets/17734314/a4543f25-97cc-4852-b7a0-15b97246e9db

---

Is there any workaround for this issue, or are there alternative approaches to select and copy the entire paginated/scrolled output?
